File: MyNotesProject/CommonLibrary/src/main/java/com/fsk/common/threads/ThreadException.java
package com.fsk.common.threads;


/**
 * An exception that documents an error with the thread.
 */
public class ThreadException extends RuntimeException {

    /**
     * Constructor.
     */
    public ThreadException () {
        super();
    }


    /**
     * Constructor.
     *
     * @param detailMessage the detail message for this exception.
     */
    public ThreadException (String detailMessage) {
        super(detailMessage);
    }


    /**
     * Constructor.
     *
     * @param detailMessage the detail message for this exception.
     * @param throwable the cause of this exception.
     */
    public ThreadException (String detailMessage, Throwable throwable) {
        super(detailMessage, throwable);
    }


    /**
     * Constructor.
     *
     * @param throwable the cause of this exception.
     */
    public ThreadException (Throwable throwable) {
        super(throwable);
    }
}

File: MyNotesProject/CommonLibrary/src/main/java/com/fsk/common/threads/ThreadException.kt
package com.fsk.common.threads;


/**
 * An exception that documents an error with the thread.
 */
class ThreadException : RuntimeException {

    /**
     * Basic Constructor.
     */
    constructor () : super();

    /**
     * Constructor.
     *
     * @param detailMessage the detail message for this exception.
     */
    constructor (detailMessage: String?) : super(detailMessage);


    /**
     * Constructor.
     *
     * @param detailMessage the detail message for this exception.
     * @param throwable the cause of this exception.
     */
    constructor (detailMessage: String?,
                 throwable: Throwable?) : super(detailMessage, throwable);


    /**
     * Constructor.
     *
     * @param throwable the cause of this exception.
     */
    constructor (throwable: Throwable?) : super(throwable);
}
